app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SignInWithSAAccountController.scala

```diff
@@ -0,0 +1,86 @@
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.auth.AuthAction
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.logging.EventLoggerService
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.logging.LoggingEvent._
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.repository.SessionKeys.REDIRECT_URL
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.repository.TEASessionCache
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.services.{EACDService, UsersGroupSearchService}
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.views.html.SignInWithSAAccount
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class SignInWithSAAccountController @Inject()(
+                                        authAction: AuthAction,
+                                        mcc: MessagesControllerComponents,
+                                        signInWithSAAccountPage: SignInWithSAAccount,
+                                        signOutController:SignOutController,
+                                        reportSuspiciousIDController: ReportSuspiciousIDController,
+                                        eacdService: EACDService,
+                                        sessionCache: TEASessionCache,
+                                        logger: EventLoggerService,
+                                        usersGroupSearchService: UsersGroupSearchService
+                                     )(implicit ec: ExecutionContext)
+  extends FrontendController(mcc) with I18nSupport {
+
+  implicit val baseLogger: Logger = Logger(this.getClass.getName)
+
+  def view(): Action[AnyContent] = authAction.async { implicit request =>
+
+    eacdService.getUsersAssignedSAEnrolment.value
+      .flatMap {
+        case Right(response)
+          if response.enrolledCredential
+            .fold(false)(_ != request.userDetails.credId) =>
+          usersGroupSearchService
+            .getAccountDetails(response.enrolledCredential.get)
+            .value
+            .map {
+              case Right(saAccountDetails) =>
+                  Ok(
+                    signInWithSAAccountPage(
+                      saAccountDetails,
+                      signOutController,
+                      reportSuspiciousIDController
+                    )
+                  )
+              case Left(_) => InternalServerError
+            }
+        case _ =>
+          logger.logEvent(
+            logIncorrectUserTypeForSASignInAgainPage(
+              request.userDetails.credId
+            )
+          )
+          sessionCache.getEntry[String](REDIRECT_URL).map {
+            case Some(redirectUrl) =>
+              Redirect(routes.AccountCheckController.accountCheck(redirectUrl))
+            case None =>
+              InternalServerError
+          }
+      }
+
+  }
```

Comment:
A POST action needs to be put in here. If you look at the LandingPage you will see how it is done :-)

Reply:
I've removed the controllers I was passing in and I have now introduced routes in the view. I also added the POST action for the button and its respective route. 👍